internal/cli/custom_domains.go

```diff
@@ -0,0 +1,319 @@
+package cli
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/auth0/auth0-cli/internal/ansi"
+	"github.com/auth0/auth0-cli/internal/auth0"
+	"github.com/auth0/auth0-cli/internal/prompt"
+	"github.com/spf13/cobra"
+	"gopkg.in/auth0.v5/management"
+)
+
+const (
+	customDomainProvisioningTypeAuth0 = "auth0_managed_certs"
+	customDomainProvisioningTypeSelf = "self_managed_certs"
+	customDomainVerificationMethodTxt = "txt"
+)
+
+var (
+	customDomainID = Argument{
+		Name: "Id",
+		Help: "Id of the custom domain.",
+	}
+
+	customDomainDomain = Flag{
+		Name:       "Domain",
+		LongForm:   "domain",
+		ShortForm:  "d",
+		Help:       "Domain name.",
+		IsRequired: true,
+	}
+
+	customDomainType = Flag{
+		Name:       "Provisioning Type",
+		LongForm:   "type",
+		ShortForm:  "t",
+		Help:       "Custom domain provisioning type. Must be 'auth0' for Auth0-managed certs or 'self' for self-managed certs.",
+	}
+
+	customDomainVerification = Flag{
+		Name:      "Verification Method",
+		LongForm:  "verification",
+		ShortForm: "v",
+		Help:      "Custom domain verification method. Must be 'txt'.",
+	}
+)
+
+func customDomainsCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "domains",
+		Short: "Manage custom domains",
+		Long:  "Manage custom domains.",
+	}
+
+	cmd.SetUsageTemplate(resourceUsageTemplate())
+	cmd.AddCommand(listCustomDomainsCmd(cli))
+	cmd.AddCommand(showCustomDomainCmd(cli))
+	cmd.AddCommand(createCustomDomainCmd(cli))
+	cmd.AddCommand(deleteCustomDomainCmd(cli))
+	cmd.AddCommand(verifyCustomDomainCmd(cli))
+
+	return cmd
+}
+
+func listCustomDomainsCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Args:    cobra.NoArgs,
+		Short:   "List your custom domains",
+		Long: `List your existing custom domains. To create one try:
+auth0 branding domains create`,
+		Example: `auth0 branding domains list
+auth0 branding domains ls`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var list []*management.CustomDomain
+
+			if err := ansi.Waiting(func() error {
+				var err error
+				list, err = cli.api.CustomDomain.List()
```

Comment:
I noticed we're not passing the context downstream, to handle cancelation properly, eventually support timeouts from the root CLI command.

this can be done with the option
https://github.com/go-auth0/auth0/blob/5e36f59f4a866da1c4858b650302c04235dc3508/management/management.go#L399-L404

the command holds its context at `cmd.Context()`

It seems we're not doing this in any of our API invocations, so feel free to ignore this for now, I can tackle the update on a separated PR.

Reply:
Yes, that would need to be done in a separate PR.